polus-unet-training-plugin/src/bfio_finetuning.py

```diff
@@ -0,0 +1,455 @@
+    dx = [ -1, -1, -1,  0 ]
+    dz = [ 0,  0,  0,  0 ]  
+    idx = -1
+    labelsData = [float(0.0) for i in range(H*W)]
+    weightsData = [float(-1.0) for i in range(H*W)]
+    samplePdfData = [float(foregroundbackgroundratio) for i in range(H*W)]
+
+
+    for x in range(H):
+        for y in range(W):
+            idx+=1
+            classLabel = classlabelsdata[idx]
+            if (classLabel == 0): 
+                weightsData[idx] = float(0.0)
+                samplePdfData[idx] = float(0.0)
+            elif (classLabel == 1):
+                pass
+            else:
+                instanceLabel = instancelabelsdata[idx]
+                nbIdx = 0
+                for nbIdx in range(len(dx)):
+                    if (z + dz[nbIdx] < 0 or y + dy[nbIdx] < 0 or y + dy[nbIdx] >= W or x + dx[nbIdx] < 0 or x + dx[nbIdx] >= H):
+                        pass
+                    else:
+                        nbInst = instancelabelsdata[(x + dx[nbIdx]) * W + y + dy[nbIdx]]
+                        if (nbInst > 0 and nbInst != instanceLabel): break
+                nbIdx+=1
+                if (nbIdx == len(dx)):
+                    labelsData[idx] = classLabel - 1
+                    weightsData[idx] = float(1.0)
+                    samplePdfData[idx] = float(1.0)
+
+    min1Dist = [BG_VALUE for i in range(H*W)]
+    min2Dist = [BG_VALUE for i in range(H*W)]
+    
+    extraWeights = np.zeros([H*W])
+    va = 1.0 - foregroundbackgroundratio
+
+    logger.info("nComponents = {}".format(nComponents))
+
+    for i in range(1, nComponents+1):
+        instancelabels = np.zeros([H, W], dtype = np.uint8)
+        instancelabels = np.reshape(np.where(instancelabelsdata == i, 0, 255), (H, W))
+        d = np.zeros([H*W])
+        dist = cv2.cv2.distanceTransform(np.uint8(instancelabels), cv2.cv2.DIST_L2, 3)
+        d = dist.flatten()
+        for j in range(H*W):
+            min1dist = min1Dist[j]
+            min2dist = min(min2Dist[j], float(d[j]))
+            min1Dist[j] = min(min1dist, min2dist)
+            min2Dist[j] = max(min1dist, min2dist)
+
+    for z in range(D):
+        for i in range(H*W):
+            if (weightsData[i] >= float(0.0)): continue
+            d1 = min1Dist[z * W * H + i]
+            d2 = min2Dist[z * W * H + i]
+            wa = math.exp(-(d1 * d1) / (2 * sigma1Px * sigma1Px))
+            we = math.exp(-(d1 + d2) * (d1 + d2) /(2 * borderWeightSigmaPx * borderWeightSigmaPx))
+            extraWeights[z * H * W + i] += borderWeightFactor * we + va * wa
+    
+    for z in range(D):
+      for i in range(H*W):
+        if (weightsData[i] >= float(0.0)): continue
+        weightsData[i] = float(foregroundbackgroundratio) + extraWeights[z * H * W + i]
+    
+    _weights = np.reshape(weightsData, (H,W))
+    _labels = np.reshape(labelsData, (H,W))
+    _samplepdf = np.reshape(samplePdfData, (H,W))
+
+    return _weights, _labels, _samplepdf
+
+
+def createlabelsandweightsfromrois(image, roiimage):
```

Comment:
Use camelCase for the function name to remain consistent with the rest of the functions defined in this file.